components/CostsPie.vue

```diff
@@ -0,0 +1,261 @@
+import accessibilityInitialize from "highcharts/modules/accessibility";
+import sunburstInitialize from "highcharts/modules/sunburst";
+import throttle from "lodash.throttle";
+import { costsPieColors, costsPieTooltipText } from "./utils/charts";
+
+const props = defineProps<{
+  hideTooltips: boolean
+  pieSize?: number
+}>();
+
+accessibilityInitialize(Highcharts);
+sunburstInitialize(Highcharts);
+
+interface pieCost {
+  id: string
+  parent: string
+  name: string
+  value?: number // If value is not provided, Highcharts calculates the children's sum itself.
+}
+
+const appStore = useAppStore();
+const chartContainerId = "costsPieContainer";
+const hideTooltipsClassName = "hide-tooltips";
+const chartBackgroundColor = "transparent";
+const chartBackgroundColorOnExporting = "white";
+let chart: Highcharts.Chart;
+let costsData: pieCost[] = [];
+
+// Prioritise showing labels for larger slices over smaller slices, where labels would otherwise overlap.
+const lowerLevelsDataLabelFilter = {
+  property: "innerArcLength",
+  operator: ">",
+  value: 16,
+};
+
+const chartLevelsOptions = (isDrillingDown: boolean = false) => [{
+  level: 1,
+  levelIsConstant: false,
+  levelSize: {
+    unit: "weight",
+    value: 1,
+  },
+}, {
+  level: 2,
+  colorByPoint: true,
+  levelSize: {
+    unit: "weight",
+    value: 2,
+  },
+  dataLabels: {
+    filter: isDrillingDown ? undefined : lowerLevelsDataLabelFilter,
+  },
+}, {
+  level: 3,
+  colorVariation: {
+    key: "brightness",
+    to: -0.75,
+  },
+  dataLabels: {
+    enabled: isDrillingDown,
+    filter: lowerLevelsDataLabelFilter,
+    style: {
+      fontSize: "0.9rem",
+    },
+  },
+  levelSize: {
+    unit: "weight",
+    value: 2,
+  },
+}];
+
+const chartSeries = () => {
+  return {
+    type: "sunburst",
+    data: costsData, // Empty at initialisation, populated later
+    name: "Root",
+    allowDrillToNode: true,
+    borderRadius: 3,
+    cursor: "pointer",
+    events: {
+      click(event) {
+        // When the clicked chart 'point' is in the second level of the sunburst,
+        // that means some drilling down or up (traversing up and down the chart) is triggered.
+        const isDrilling = event.point.node.level !== 1 && event.point.node.children.length > 0; // No drill-down if no children to drill down to;
+        if (!isDrilling) {
+          return;
+        }
+        const isDrillingUp = event.point.drillId === appStore.totalCost?.id;
+        const isDrillingDown = isDrilling && !isDrillingUp;
+        chart.update({ series: { levels: chartLevelsOptions(isDrillingDown) } });
+      },
+    },
+    dataLabels: {
+      rotation: 0,
+      rotationMode: "auto", // Without this, labels sometimes appear in the top left at random. https://github.com/highcharts/highcharts/issues/18953
+      format: "{point.name}",
+      style: {
+        fontSize: "1.1rem", // TODO: Make font-size smaller on small screens (or disappear on very small)
```

Comment:
Still planning on doing this TODO? One for another ticket?

Reply:
Another ticket